src/common/TrackedOp.cc

```diff
@@ -169,24 +169,22 @@ bool OpTracker::check_ops_in_flight(std::vector<string> &warning_vector)
       utime_t oldest_op_tmp = sdata->ops_in_flight_sharded.front()->get_initiated();
       if (oldest_op_tmp < oldest_op) {
         oldest_op = oldest_op_tmp;
+	if (now - oldest_op > complaint_time)
+	  break;
       }
     } 
-    total_ops_in_flight += sdata->ops_in_flight_sharded.size();
   }
       
-  if (0 == total_ops_in_flight)
+  if (i == num_optracker_shards)
```

Comment:
I don't really understand how switching from explicitly counting the number of ops to seeing if we made it through the whole list without breaking is a code cleanup. What's the purpose?
In fact I like the old way better: it would short-circuit if there were no ops, but otherwise it would log the oldest one. With this change it won't log the oldest one unless it's too old, and that means that looking at the logging data can be less informative (we don't have confirmation that things are good).